gliner/model.py

```
@@ -992,6 +1003,93 @@ def unfreeze_component(self, component_name: str):
+        
+        Raises:
+            ImportError: If the `peft` library is not installed.
+            RuntimeError: If the encoder backbone cannot be located at
+                `self.model.token_rep_layer.bert_layer.model` and LoRA cannot be applied.
+        """
+        try:
+            from peft import TaskType, LoraConfig, get_peft_model
+        except ImportError:
+            raise ImportError(
+                "peft is required for LoRA training. Install with: pip install peft"
+            )
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)